tests/test_tempfile.py

```diff
@@ -0,0 +1,185 @@
+from __future__ import annotations
+
+import os
+import pathlib
+import shutil
+import tempfile
+
+import pytest
+
+from anyio import (
+    NamedTemporaryFile,
+    SpooledTemporaryFile,
+    TemporaryDirectory,
+    TemporaryFile,
+    gettempdir,
+    gettempdirb,
+    gettempprefix,
+    gettempprefixb,
+    mkdtemp,
+    mkstemp,
+    to_thread,
+)
+
+pytestmark = pytest.mark.anyio
+
+
+@pytest.mark.anyio
+async def test_temporary_file() -> None:
+    data = b"temporary file data"
+    async with TemporaryFile[bytes]() as af:
+        await af.write(data)
+        await af.seek(0)
+        result = await af.read()
+    assert result == data
+    assert af.closed
+
+
+@pytest.mark.anyio
+async def test_named_temporary_file() -> None:
+    data = b"named temporary file data"
+    async with NamedTemporaryFile[bytes]() as af:
+        filename: str = str(af.name)
```

Comment:
Looks like a redundant cast?
```suggestion
        filename = str(af.name)
```